Missions/__AR/11_cont/MAIN.src

```diff
@@ -1023,6 +1027,43 @@ Export Function CreateUnitsWithClass(num_units,un_class,side);
                     result = result ^ CreateHuman;
                end;
      end;
+Export Function CreateUnitsWithClassLimited(num_units,un_class,side,limit);
```

Comment:
consider a new function.

`CreateLimitedUnitsWithClasses(unit_numbers_list, unit_classes_list, side, limit)`
idea is to create all units in bulk, making them on a one-by-one basis rather than the current approach - which is bound to hit the problem outlined in the PRs OP.